### PR TITLE
transaction docs update

### PR DESCRIPTION
**Pull request by ines-toupeira**

### CODE REVIEW

File: learn/protocol/transactions.md [hunk 1]
```diff
@@ -1,49 +1,124 @@
...
+| `validity_start_height` | `u32` | The block height from which the transaction becomes valid. A transaction can be submitted before this height, but will only be eligible for inclusion once the blockchain reaches this height. If not included in a block within the window of 120 blocks after, the transaction will expire |
+| `network_id` | `NetworkId` | Specifies the network (mainnet, testnet) on which the transaction is valid to ensure the transaction is only processed on the correct network |
+| `flags` | `TransactionFlags` | Flags that indicate special transaction types, such as contract creation or signaling |
+| `proof` | `Vec<u8>` | A cryptographic proof, generated by signing the transaction data with the sender’s private key, which authenticates the sender |
 
-Hence, when a validator includes a transaction to the micro block, it may succeed or fail, depending on the sender’s balance:
+A transaction must pass validity checks before being processed. Once the transaction is successfully validated, the boolean variable `valid` is set to `true`, confirming that the transaction has been authenticated and is ready to be included in the blockchain.
```
Comment (ii-cruz) [inline learn/protocol/transactions.md]: I think this is a bit confusing so perhaps let's not say anything here. The paragraph right before Transaction Structure refers to it needing to be validated and then refers to the mempool documentation for more info and later on doc we explain that we commit failed transactions and reject the ones that cannot be deducted their fee, seams complete enough.

File: learn/protocol/transactions.md [hunk 2]
```diff
@@ -1,49 +1,124 @@
...
 
-- **Succeed**: the sender has enough funds to cover both the transaction value and fees, and the user’s balance is updated.
-- **Failed**: the sender has insufficient funds to pay out the transaction value but transaction fees can be deducted.
+**Additional Considerations**
 
-Moreover, if the sender lacks sufficient funds to cover the transaction fee, the transaction is invalid and will not be processed.
+- **Staking-Related Transactions**: In staking-related transactions, the **sender** and **recipient** addresses can be the same, which is a specific feature of transactions interacting with the staking contract
```
Comment (ii-cruz) [inline learn/protocol/transactions.md]: This cannot happen as we discussed. Just making it so you see where it is.

File: learn/protocol/transactions.md [hunk 3]
```diff
@@ -1,49 +1,124 @@
...
+- **Variations in Transaction Structure**: Different account types (basic, HTLC, vesting, staking) may have variations in their transaction structure. These variations ensure that each account type processes transactions according to its purpose and function within the protocol
+- **Incoming and Outgoing Transactions**: The network distinguishes between incoming and outgoing transactions. **Incoming transactions** are related to operations within the blockchain such as adding stake or creating an account, while **outgoing transactions** involve external operations that affect the network state, such as deleting a validator or removing stake
```
Comment (ii-cruz) [inline learn/protocol/transactions.md]: As we discussed, this explanation needs some corrections. Marking it so you don't forget

File: learn/protocol/transactions.md [hunk 4]
```diff
@@ -1,49 +1,124 @@
...
+- **Successful**: the sender has enough funds to cover both the transaction value and fees, and the user’s balance is updated
+- **Failed**: the sender has insufficient funds to pay out the transaction value but transaction fees can be deducted
+
+**Example Scenario**
+
+Consider a scenario where Alice has 100 NIM in her balance. She sends 3 transactions:
+
+- **Transaction 1**: Alice sends 80 NIM to Bob. Since she has sufficient funds (80 NIM + fees), this transaction succeeds, and both the value and fees are deducted from her account
+- **Transaction 2**: Alice attempts to send 50 NIM to Charlie immediately after. Since her remaining balance is insufficient to cover both the value and fees, this transaction fails, but the transaction fees are still deducted
+- **Transaction 3**: Alice tries to send another transaction but does not have enough NIM to cover even the fees. In this case, the transaction is deemed **invalid** and is not processed by the network. Neither the value nor the fees are deducted
```
Comment (ii-cruz) [inline learn/protocol/transactions.md]:
```suggestion
- **Transaction 3**: Alice tries to send another transaction but does not have enough NIM to cover the fee. In this case, the transaction is deemed **invalid** and is not processed by the network. Neither the value nor the fees are deducted
```

File: learn/protocol/transactions.md [hunk 5]
```diff
@@ -1,49 +1,124 @@
...
+
+## Transaction Finality
+
+Finality is periodically reinforced through the Tendermint protocol, which ensures that blocks and the transactions within them cannot be reversed with once they are finalized by a macro block. For more details on how finality is achieved through the block structure, please refer to the [block format](block-format.md) documentation.
```
Comment (ii-cruz) [inline learn/protocol/transactions.md]:
```suggestion
Finality is periodically reinforced through the Tendermint protocol, which ensures that blocks and the transactions within them cannot be reversed once they are finalized by a macro block. For more details on how finality is achieved through the block structure, please refer to the [block format](block-format.md) documentation.
```

Comment (ii-cruz) [reply learn/protocol/transactions.md]: Do you mean block structure here? It seems like the finality is achieved by the structure of a macro block and I guess what you mean is that the finality is periodically achieved by the consensus algorithm we use.

File: learn/protocol/transactions.md [hunk 6]
```diff
@@ -1,49 +1,124 @@
...
+
+## Inherents
+
+In addition to user-initiated transactions, Nimiq also processes **inherents**—system-generated operations that modify the blockchain’s state without requiring a transaction from a user. These are used for tasks like distributing rewards and enforcing penalties, ensuring network stability.
```
Comment (ii-cruz) [inline learn/protocol/transactions.md]:
```suggestion
In addition to user-initiated transactions, Nimiq also processes **inherents**—system-generated operations that modify the blockchain’s state without requiring a transaction from a user. These are used for tasks like distributing rewards and enforcing punishments, ensuring network stability.
```

File: learn/protocol/transactions.md [hunk 7]
```diff
@@ -1,49 +1,124 @@
...
+- **No Account Interaction**: Inherents don’t originate from user accounts or affect balances
+- **System-Driven**: These operations are generated by the protocol itself based on network rules and conditions
+- **No Signature Requirement**: Inherents don’t require digital signatures for validation.
+
+There are 5 types of inherents in the Nimiq protocol:
+
+- **Reward**: Automatically issued to validators who successfully fulfill their assigned slots without misbehavior
+- **Penalty**: Applied to validators who delay block production. The penalty removes their reward for the slot, and the validator can be deactivated
+- **Jail**: Enforced when validators are involved in severe misbehavior, such as double voting or creating forks. In this case, all of their slots are punished, and the validator is jailed, preventing further participation in the network for 8 epochs
+- **Finalize Batch**: Triggered at the end of each batch of micro blocks, marking the completion of a micro block batch
```
Comment (ii-cruz) [inline learn/protocol/transactions.md]:
```suggestion
- **Finalize Batch**: Triggered at the end of each batch of micro blocks, marking its completion
```

**Review by ii-cruz — CHANGES_REQUESTED** [review]

File: learn/protocol/transactions.md [hunk 8]
```diff
@@ -1,49 +1,123 @@
...
+### Transaction Builder Errors
+
+Deals with missing or invalid fields when creating transaction. If any of the following errors occur, the transaction will not even be sent:
+
+- **No Sender**: The sender address is missing from the transaction
+- **No Recipient**: The recipient address is missing. A valid recipient must be provided
+- **No Value**: The transaction value (amount to be transferred) is missing
+- **No Validity Start Height**: The transaction’s validity start height, which defines from when the transaction becomes valid, is not set
+- **No Network ID**: The transaction’s network ID is missing. The ID identifies which network the transaction is intended, whether Testnet or Mainnet
+- **Invalid Sender**: The sender is not valid for the recipient. Some transactions, particularly staking-related ones, have special restrictions on the sender field. For example, self-transactions in the staking contract require the sender to be the same as the recipient. This error occurs if these conditions are not met (retire and reactivate)
```
Comment (ii-cruz) [inline learn/protocol/transactions.md]: The sender and recipient cannot be the same

File: learn/protocol/transactions.md [hunk 9]
```diff
@@ -1,49 +1,123 @@
...
+
+### Transaction Error
+
+Deals with transaction execution issues once it is submitted to the network:
+
+- **Foreign Network**: The transaction is intended for a network other than the one it was submitted to (a transaction for the Testnet being submitted to the Mainnet)
+- **Zero Value**: The transaction’s value is set to zero when it’s not allowed. Transactions that transfer funds must have a non-zero value unless they are specific contract interactions that permit zero-value transactions
+- **Invalid Value**: The value of the transaction is not valid for the specific operation being attempted (incorrect value for contract-related transactions)
+- **Overflow**: The transaction causes an overflow in calculations, due to large values that cannot be processed
+- **Sender Equals Recipient**: The sender and recipient of the transaction are the same, which is not allowed unless under specific conditions (e.g., in staking-related transactions)
```
Comment (ii-cruz) [inline learn/protocol/transactions.md]: The sender and recipient cannot be the same. The staking contract is no exception

**Review by ii-cruz — APPROVED** [review]
LGTM :)

**Review by paberr — APPROVED** [review]

**Pull request merged**